src/simple.rs

```diff
@@ -0,0 +1,275 @@
+///   if unsure)
+/// - `message` is a message from which the digest is to be
+///   calculated, it can be empty.
+/// - The function returns the message digest.
+///
+/// [`DigestAlgorithm::Sha256`]: enum.DigestAlgorithm.html#variant.Sha256
+pub fn simple_calculate_digest(algo: &DigestAlgorithm, message: &[u8]) -> Result<Vec<u8>>
+{
+    let algo = conv::digest_rs_to_c(algo);
+    let message_len = message.len();
+    let message = message.as_ptr() as *const c_char;
+    let mut digest = ptr::null();
+    let mut digest_len = 0;
+    let r = unsafe {
+        lib::yaca_simple_calculate_digest(algo, message, message_len,
+                                          &mut digest, &mut digest_len)
+    };
+    conv::res_c_to_rs(r)?;
+    Ok(common::vector_from_raw(digest_len, digest))
+}
+
+/// Creates a signature using asymmetric private key
+///
+/// - `algo` is a digest algorithm that will be used.
+/// - `prv_key` is a private key that will be used, algorithm is
+///   deduced based on key type, supported key types:
+///   * [`KeyType::RsaPrivate`],
+///   * [`KeyType::DsaPrivate`],
+///   * [`KeyType::EcPrivate`].
+/// - `message` is message to be signed, it can be empty.
+/// - The function returns a message signature.
+/// - For [`DigestAlgorithm::Sha384`] and
+///   [`DigestAlgorithm::Sha512`] the `RSA` key size must be bigger
+///   than 512 bits.
+/// - Using [`DigestAlgorithm::Md5`] algorithm for `DSA` and
+///   `ECDSA` operations is prohibited.
+///
+/// [`KeyType::RsaPrivate`]: enum.KeyType.html#variant.RsaPrivate
+/// [`KeyType::DsaPrivate`]: enum.KeyType.html#variant.DsaPrivate
+/// [`KeyType::EcPrivate`]: enum.KeyType.html#variant.EcPrivate
+/// [`DigestAlgorithm::Sha384`]: enum.DigestAlgorithm.html#variant.Sha384
+/// [`DigestAlgorithm::Sha512`]: enum.DigestAlgorithm.html#variant.Sha512
+/// [`DigestAlgorithm::Md5`]: enum.DigestAlgorithm.html#variant.Md5
+pub fn simple_calculate_signature(algo: &DigestAlgorithm, prv_key: &Key,
+                                  message: &[u8]) -> Result<Vec<u8>>
+{
+    let algo = conv::digest_rs_to_c(algo);
+    let prv_key = key::get_handle(&prv_key);
+    let message_len = message.len();
+    let message = message.as_ptr() as *const c_char;
+    let mut signature = ptr::null();
+    let mut signature_len = 0;
+    let r = unsafe {
+        lib::yaca_simple_calculate_signature(algo, prv_key, message, message_len,
+                                             &mut signature, &mut signature_len)
+    };
+    conv::res_c_to_rs(r)?;
+    Ok(common::vector_from_raw(signature_len, signature))
+}
+
+/// Verifies a signature using asymmetric public key
+///
+/// - `algo` is a digest algorithm used to calculate the signature.
+/// - `pub_key` is a matching public key to the one used to
+///   calculate the signature, algorithm is deduced based on key
+///   type, supported key types:
+///   * [`KeyType::Rsa(Public)`],
+///   * [`KeyType::Dsa(Public)`],
+///   * [`KeyType::Ec(Public)`].
```

Comment:
Broken links.

Is there an option to ```cargo doc``` that treats those as hard errors?

Reply:
You're the master of rust lints, you tell me :-D

Reply:
> broken_intra_doc_links
> 
> This lint warns by default and is nightly-only. This lint detects when an intra-doc link fails to get resolved. For example:

Please, check this. I'm not on nightly.

Reply:
Doesn't seem to work on current nightly (forced it, happily proceeded to compile & build the doc w/ dead links). Oh well.

Reply:
Anything else here then?

Reply:
Not that I've noticed, resolving.